EOLib/PacketHandlers/Guild/GuildTakeHandler.cs

```diff
@@ -0,0 +1,38 @@
+﻿using AutomaticTypeMapper;
+using EOLib.Domain.Interact;
+using EOLib.Domain.Interact.Guild;
+using EOLib.Domain.Login;
+using EOLib.Domain.Map;
+using EOLib.Net.Handlers;
+using Moffat.EndlessOnline.SDK.Protocol.Net;
+using Moffat.EndlessOnline.SDK.Protocol.Net.Server;
+using Optional;
+using System.Collections.Generic;
+
+namespace EOLib.PacketHandlers.Guild
+{
+    [AutoMappedType]
+
+    public class GuildTakeHandler : InGameOnlyPacketHandler<GuildTakeServerPacket>
+    {
+        private readonly IGuildSessionRepository _guildSessionRepository;
+
+        public override PacketFamily Family => PacketFamily.Guild;
+
+        public override PacketAction Action => PacketAction.Take;
+
+        public GuildTakeHandler(IPlayerInfoProvider playerInfoProvider,
+                                 IGuildSessionRepository guildSessionRepository)
+            : base(playerInfoProvider)
+        {
+            _guildSessionRepository = guildSessionRepository;
+        }
+
+        public override bool HandlePacket(GuildTakeServerPacket packet)
+        {
+            _guildSessionRepository.GuildDescription = packet.Description;
```

Comment:
It's unclear to me if this description is the player's current guild description or if it's the description of a guild requested via the "guild lookup" function.

If this is a search result, it shouldn't be in the guild session repository state - guild session repository should only be the current player's guild info. Otherwise this is fine as-is, but it'd be nice to clarify the member name slightly so it's easier to tell whose guild this description belongs to.

Reply:
This is the players guild description not a lookup